### PR TITLE
Add a quoting API inspired by Ppx_deriving

### PR DESCRIPTION
**Pull request by rgrinberg**
This PR introduces a hygiene API in the spirit of Ppx_deriving.quoter.
The main differences are:

* We can't use metaquot
* There are no optional arguments
* We don't wrap stuff with modules as this can be easily handled
  separately

### CODE REVIEW

**Comment by rgrinberg** [comment]
It seems like I'm unable to `#install_printer` on the `Pprintast` printers. Any idea as to why this isn't working?

File: src/quote.mli [hunk 1]
```diff
@@ -0,0 +1,9 @@
+open Import
```
Comment (diml) [inline src/quote.mli]: This mli needs documentation, otherwise it's not easy to understand what it does without looking at the implementation

**Comment by diml** [comment]
Not sure about the install_printer problem. We should switch all these tests to ppx_expect at some point anyway

File: src/quote.mli (hunk 1 above)
Comment (rgrinberg) [reply src/quote.mli]: Yup, I will write some documentation today.

**Comment by rgrinberg** [comment]
I've written up some docs. Shall I rename the module to `Quoter`? I'm not sure if `Quote` or `Quoter` is more natural here.

**Comment by diml** [comment]
`Quoter` seems like a better name indeed. Appart from that, it looks good to merge.

**Comment by rgrinberg** [comment]
@diml I renamed it.

**Comment by diml** [comment]
And for people who might be watching this PR: we are adding this module to make it easier to port plugins from `Ppx_deriving` to `Ppx.Deriving`

**Comment by rgrinberg** [comment]
Thanks for the review, I'll prepare a corresponding PR to ppx.

**Pull request merged**